pulsar-zookeeper-utils/src/main/java/com/yahoo/pulsar/zookeeper/ZooKeeperChildrenCache.java

```diff
@@ -88,6 +91,12 @@ public void unregisterListener(ZooKeeperCacheListener<Set<String>> listener) {
     @Override
     public void process(WatchedEvent event) {
         LOG.debug("[{}] Received ZooKeeper watch event: {}", cache.zkSession.get(), event);
-        cache.process(event, this);
+        if (!isShutdown.get()) {
+            cache.process(event, this);
+        }
+    }
+
+    public void shutdown() {
```

Comment:
Can we rename it to close() or something else? The name could confuse us since we call shutdown while stopping the service.